keep-ui/app/settings/add-user-modal.tsx

```diff
@@ -48,190 +43,163 @@ const roleOptions: RoleOption[] = [
+          // if 'detail' in errorData:
+          if (errorData.detail) {
+            setError('apiError', { type: 'manual', message: errorData.detail });
+          }
+          else{
+            setError('apiError', { type: 'manual', message: errorData.message || 'Failed to add user' });
+          }
 
-    // Make the API call to add the user
-    const response = await fetch(`${getApiURL()}/settings/users`, {
-      method: "POST",
-      headers: {
-        Authorization: `Bearer ${accessToken}`,
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify({ email, role: selectedRole, password }),
-    });
+        }
+      } catch (error) {
+        setError('apiError', { type: 'manual', message: 'An unexpected error occurred' });
+      }
+    };
 
-    if (response.ok) {
-      const newUser = await response.json();
-      setUsers((currentUsers) => [...currentUsers, newUser]);
-      // Reset form and close modal on successful addition
-      handleOnClose();
-    } else {
-      const errorData = await response.json();
-      setAddUserError(
-        errorData.message ||
-          errorData.detail ||
-          "Failed to add user (unknown error)"
-      );
-    }
-  };
+    const handleClose = () => {
+        clearErrors('apiError');
+        reset();
+        onClose();
+      };
 
-  const handleOnClose = () => {
-    onClose();
-    setEmail("");
-    setSelectedRole("");
-    setPassword("");
-    setAddUserError("");
-  };
-
-  const handleEmailChange = (email: string) => {
-    setEmail(email);
-    if (validateEmail(email)) {
-      setEmailError("");
-    } else {
-      setEmailError("Please enter a valid email address.");
-    }
-  };
+    return (
+      <Dialog as="div" className="fixed inset-0 z-10 overflow-y-auto" open={isOpen} onClose={handleClose}>
+        <div className="flex items-center justify-center min-h-screen">
+          <Dialog.Panel className="bg-white p-4 rounded" style={{ width: "400px", maxWidth: "90%" }}>
+            <Dialog.Title>Add User</Dialog.Title>
+            <form
+                onSubmit={e => {
+                    clearErrors()
+                    handleSubmit(onSubmit)(e)
+                }}
+            >
+              {/* Email/Username Field */}
+              {authType === AuthenticationType.MULTI_TENANT ? (
+                <div className="mt-4">
+                  <Subtitle>Email</Subtitle>
+                  <Controller
+                    name="email"
+                    control={control}
+                    rules={{ required: 'Email is required', pattern: { value: /\S+@\S+\.\S+/, message: 'Invalid email format' } }}
+                    render={({ field }) => (
+                        <>
+                          <TextInput {...field} error={!!errors.email} errorMessage={errors.email && typeof errors.email.message === 'string' ? errors.email.message : undefined}/>
+                        </>
+                      )}
+                  />
```

Comment:
The `Controller` component is helpful for complex data inputs. I think for the inputs, you can get away with using the [`register` object](https://react-hook-form.com/docs/useform/register) that you pass as props:

```ts
<TextInput 
  error={!!errors.email} 
  errorMessage={errors.email && typeof errors.email.message === 'string' ? errors.email.message : undefined}
  {...register("email"} 
/>
```